src/main/java/algorithms/curated170/hard/ConfusingNumber2.java

```diff
@@ -0,0 +1,38 @@
+package algorithms.curated170.hard;
+
+import algorithms.hard.StrobogrammaticNumber3;
+
+ 
+public class ConfusingNumber2 {
+   final char[][] reflectives = new char[][]{
+        {'0', '0'},
+        {'6', '9'},
+        {'1', '1'},
+        {'9', '6'},
+        {'8', '8'}
+    };
+    
+    public int confusingNumberII(int n) {
+        String num = Integer.toString(n);
+        int res = findTotal(num);
+        StrobogrammaticNumber3 strobogrammaticNumber = new StrobogrammaticNumber3();
+        return res - strobogrammaticNumber.strobogrammaticInRange("1", num) - 1;
+    }
+    private int findTotal(String s) {
+        if (s.length() == 0) return 1;
+        char first = s.charAt(0);
+        int res = count(first) * (int) (Math.pow(5, s.length() - 1));
```

Comment:
What is being done here?

Reply:
findTotal is to calculate the total count of the numbers which can be generated using those five numbers: 0, 1, 6, 8, 9.
It is why we use 5 to calculate the permutation.

As the "s" is a limit for us, we have to calculate numbers less than "s".
In order to do that: count(first) returns the numbers less than the parameter (first) from this set: 0, 1, 6, 8, 9.

findTotal is recursive as we should do the calculation for each digit.

Reply:
When considering a number like 9999, we ignore the permutations that start with 9. But, when we calculate the next digits, we are actually calculating the number of permutations that start with 9. Pretty clear now.